### ISSUE TITLE
Deserializing objects like Duration causes reads outside memory bounds

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Wire#332]
Reproduce:

```
    static final class DurationHolder extends SelfDescribingMarshallable {
        int foo;
        Duration duration;
    }

    @Test
    public void durationHolderMarshallable() {
        final DurationHolder holder = new DurationHolder();
        holder.foo = 1;
        holder.duration= Duration.ofSeconds(63);

        final String h = Marshallable.$toString(holder);
        System.out.println("h = " + h);
        // This causes Long access outside memory bounds
        final DurationHolder deserialized = Marshallable.fromString(h);
        System.out.println("deserialized = " + deserialized);

    }

```

When deserializing the holder, deserializing of the Duration field initially fails (problem 1), this causes the deserializer to retry deserialization in a catch branch. The latter deserialization will try to deserialize Duration binary which causes it to exceed the memory bounds (problem 2).

A fix should address both problem types.

**Comment by minborg** [comment]
Ideally, known immutables could be created via constructors or static create methods as opposed to reflection.

**Comment by peter-lawrey** [comment]
Java types are not assumed to have a default value and so use `null` as the default.
However, the code copying the default value wasn't checking for this is when a copy of `null` was used using unsafe it blew up.
I have changed the code to both check for such a null value and support Duration properly.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.22ea9](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.22ea9), [BOM-2.22ea49](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea49)
